### ISSUE TITLE
Use semantic versioning and separate versioning for all packages

### ISSUE THREAD

**Issue by lenton** [kohana/kohana#59]
I was surprised to find out that Kohana doesn't strictly follow semantic versioning (http://semver.org). I propose we following this specification when versioning all of Kohana's packages.

I also propose that each package should have its own separate versioning. Currently, when releasing a new version of Kohana we release every single module to the same version (even if it hasn't even been modified!).

**Comment by enov** [comment]
I am mostly :+1: on this. 

The overall versioning should be fixed.

However, I am not 100% sure regarding each package having its own separate versioning. At this point they are tightly coupled, and when we say kohana/cache v.3.3 it means "the cache version that should be used along with kohana/core v.3.3".

Here are, by the way, some of the discussions from our old issue tracker.

http://dev.kohanaframework.org/issues/4477
http://dev.kohanaframework.org/issues/4160

**Comment by lenton** [comment]
> However, I am not 100% sure regarding each package having its own separate versioning. At this point they are tightly coupled, and when we say kohana/cache v.3.3 it means "the cache version that should be used along with kohana/core v.3.3".

I would disagree with the modules being tightly coupled, I think most modules are pretty much solitary. Finding which modules work with what only becomes a problem when they depend on each other and even then, the versions will be specified in the composer config.

> http://dev.kohanaframework.org/issues/4160

Wow, Zeelot has already described this issue in great detail.

If we truely want decoupled modules then I feel this is a step that must be taken.

**Comment by rjd22** [comment]
Since we won't be supporting all the modules anymore and we can enforce version restraints we composer I'm all for this. 

This will enable us to move faster and make releasing stuff more easy. If there is a coupling problem we should fix this by making the modules less dependent on the framework.
